### PR TITLE
95/t/TECH-900/Honey Badger Integration Phase 2

### PR DESCRIPTION
**Pull request by ibiziiac-sv**
WIP

### CODE REVIEW

**Comment by ibiziiac-sv** [comment]
@codyjca can you please make an early CR? 

Remaining work: 
* setup `Honeybadger.exception_flter` to send to honeybadger.io only the exceptions handled by `ExceptionHandling`.

File: test/unit/exception_handling_test.rb [hunk 1]
```diff
@@ -360,6 +360,26 @@ def log_error_callback(data, ex)
...
+          end
+        end
+      end
+
+      should "invoke send_exception_to_honeybadger when log_error is executed" do
+        ExceptionHandling.expects(:send_exception_to_honeybadger).times(1)
+        ExceptionHandling.log_error(exception_1)
+      end
+
+      should "send exception to Honeybadger when send_exception_to_honeybadger is executed" do
```
Comment (codyjca) [inline test/unit/exception_handling_test.rb]: Can you add a test to make sure Honeybadger.notify is not invoked when we call log_error but Honeybadger is not defined?

File: lib/exception_handling.rb [hunk 2]
```diff
@@ -202,6 +204,8 @@ def log_error(exception_or_string, exception_context = '', controller = nil, tre
 
         write_exception_to_log(ex, exception_context, timestamp)
 
+        send_exception_to_honeybadger(ex)
```
Comment (codyjca) [inline lib/exception_handling.rb]: Worth bringing up in the demo, we but might want to include exception_context and timestamp if we find those useful.

**Comment by ibiziiac-sv** [comment]
@codyjca CR comments addressed. Please have a look again

**Comment by codyjca** [comment]
Looks good to me :+1:, only thing that I can think of coming up is if `File.basename(caller_path) == "exception_handling.rb"` is the best way to determine if the exception comes from exception handling vs honeybadger. Demo should prove this though.

I wonder if it's worth putting in a test for .ensure_safe? I know that was one of the other exceptions that we wanted to confirm (however I see that you put in the test at a lower level so might not be needed, I'll let you decide)

**Comment by ibiziiac-sv** [comment]
@codyjca thanks for the review. I will add a new test for `ensure_safe`. I am testing it now and it seems that `Honeybadger.exception_filter` has to be placed somewhere at the initialization level, because having it in the exception handling gem, the filter works too late and we get duplicated logs notifications on honeybadger.io

**Comment by ibiziiac-sv** [comment]
@codyjca it seems that there's a better way to filter out the exceptions according to HB documentation: http://docs.honeybadger.io/lib/ruby.html#configuration
`exceptions.enabled` (Boolean): Enable automatic reporting of exceptions. Default: `true` 
So if I disable the automatic reporting and only report them manually with `Honeybadger.notify`, we don't need to use `Honeybadger.exception_filter` anymore.

**Pull request merged**